src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -133388,26 +133867,36 @@ public void LambdaNullReturn_03()
             var source =
 @"#nullable enable
 using System;
+using System.Threading.Tasks;
 class Program
 {
-    static void F<T>(Func<T> f)
+    static void F<T>(Func<Task<T>> f)
     {
     }
-    static void M<T>(T? t) where T : class
+    static void M1<T>(T? t) where T : class
     {
-        F(() =>
+        F(async () =>
         {
+            await Task.Yield();
+            if (t is null) return null;
+            return t;
+        });
+    }
+    static void M2<T>(T? t) where T : class?
+    {
+        F(async () =>
+        {
+            await Task.Yield();
             if (t is null) return null;
             return t;
         });
     }
 }";
-            var comp = CreateCompilation(source);
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
             comp.VerifyEmitDiagnostics();
         }
```

Comment:
The only changes in `LambdaNullReturn_01` to `_06` is that each pair of tests have been combined. Two new tests `04` and `_05` were added at the end. (There were already too many similar tests and adding 4 news tests rather than combining pairs would have made the situation worse.)